cdap-common/src/main/java/co/cask/cdap/common/security/AuthEnforceUtil.java

```diff
@@ -37,17 +53,106 @@ private AuthEnforceUtil() {
     // no-op
   }
 
+  private static final Map<Class<? extends EntityId>, Constructor<? extends EntityId>> CONS_CACHE =
+    new ConcurrentHashMap<>();
```

Comment:
This seems overly complicated. Since you have a fixed set of `EntityId` to support, simply initialize a `Map` in a static initialization block `static { }` should be good enough.